ipam/allocator.go

```diff
@@ -708,6 +711,15 @@ func (alloc *Allocator) update(sender mesh.PeerName, msg []byte) error {
 		}
 	}
 
+	if data.Ring == nil && data.Paxos == nil {
+		// Neither Ring nor Paxos present - sender is a non-elector
+		// waiting for a ring update. Send them a copy of ours if we
+		// have one
+		if !alloc.ring.Empty() && sender != mesh.UnknownPeerName {
+			alloc.sendRingUpdate(sender)
+		}
+	}
+
```

Comment:
I think we want to replace the three top-level conditionals with a three-branch `switch`.